cookbook/case_studies/ml_training/spark_horovod/keras_spark_rossmann_estimator.py

```diff
@@ -490,18 +548,25 @@ def act_sigmoid_scaled(x):
     else:
         K.set_session(tf.Session(config=tf.ConfigProto(device_count={"GPU": 0})))
 
-    # build the model
+    # Model compiling
+    # ================
+    #
+    # build the Keras model
     inputs = {col: Input(shape=(1,), name=col) for col in all_cols}
     embeddings = [
         Embedding(len(vocab[col]), 10, input_length=1, name="emb_" + col)(inputs[col])
         for col in CATEGORICAL_COLS
     ]
+    # we define a layer that concatenates the list of inputs
     continuous_bn = Concatenate()(
         [Reshape((1, 1), name="reshape_" + col)(inputs[col]) for col in CONTINUOUS_COLS]
     )
+    # we define a layer to normalize its inputs
     continuous_bn = BatchNormalization()(continuous_bn)
     x = Concatenate()(embeddings + [continuous_bn])
+    # without affecting the batch size of data, it flattens the input
```

Comment:
Can we have all the comments you have included in the model at the top, before defining the model?